test/lib/ansible_test/_util/target/sanity/import/importer.py

```diff
@@ -191,9 +221,36 @@ def find_module(self, fullname, path=None):
             # not a namespace we care about
             return None
 
+        def create_module(self, spec):  # pylint: disable=unused-argument
+            """Return None to use default module creation.
+            :param spec: ModuleSpec
+            """
+            return None
+
+        def exec_module(self, module):
+            """Execute the module if the name is ansible.module_utils.basic and otherwise raise an ImportError
+            :param module: module
+            """
+            fullname = module.__spec__.name
+            if fullname == 'ansible.module_utils.basic':
+                self.loaded_modules.add(fullname)
+                for path in convert_ansible_name_to_absolute_paths(fullname):
+                    if not os.path.exists(path):
+                        continue
+                    loader = SourceFileLoader(fullname, path)
+                    spec = spec_from_loader(fullname, loader)
+                    real_module = module_from_spec(spec)
+                    loader.exec_module(real_module)
+                    real_module.AnsibleModule = ImporterAnsibleModule
+                    real_module._load_params = lambda *args, **kwargs: {}  # pylint: disable=protected-access
+                    sys.modules[fullname] = real_module
+                    return
+            raise ImportError('import of "%s" is not allowed in this context' % fullname)
```

Comment:
Just to be safe, we should raise an exception here if `ansible.module_utils.basic` was not found. It shouldn't happen, but if it does, the `ImportError` message will be wrong and confusing.

Reply:
I'm confused why this error should'nt be the same as the Py2 codepath. Is ImportError also wrong in that case?

Reply:
I think he's suggesting a second error handler- currently this fall-through would be used for both "you tried to exec_module something that wasn't ansible.module_utils.basic" (for which the current message is fine), but also the case where the for loop completed without finding basic (in which case the error message will be incorrect and confusing). Basically just add an `else` clause to the end of the `for path in convert...` loop that raises a different ImportError that we couldn't find basic.